core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -379,21 +365,24 @@ public boolean containsMountPoint(AlluxioURI uri, boolean containsSelf)
    */
   public List<MountInfo> findChildrenMountPoints(AlluxioURI uri, boolean containsSelf)
       throws InvalidPathException {
-    String path = uri.getPath();
-    List<MountInfo> childrenMountPoints = new ArrayList<>();
-
     try (LockResource r = new LockResource(mReadLock)) {
-      for (Map.Entry<String, MountInfo> entry : mState.getMountTable().entrySet()) {
-        String mountPath = entry.getKey();
-        if (!containsSelf && mountPath.equals(path)) {
-          continue;
-        }
-        if (PathUtils.hasPrefix(mountPath, path)) {
-          childrenMountPoints.add(entry.getValue());
-        }
-      }
+      return getChildrenMountPointsStream(uri, containsSelf).map(TrieNode::getValue)
+          .collect(Collectors.toList());
     }
-    return childrenMountPoints;
+  }
+
+  private Stream<TrieNode<MountInfo>> getChildrenMountPointsStream(AlluxioURI uri,
```

Comment:
i'm actually thinking if this stream abstract is necessary, compared to a vanilla for loop. WDYT?